model/task_lifecycle.go

```diff
@@ -576,6 +592,14 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		}
 		err = b.SetCachedTaskFinished(t.Id, status, &t.Details, t.TimeTaken)
 		if err != nil {
+			grip.Error(message.Fields{
+				"message":    "failed to set cached task finished",
+				"function":   "UpdateBuildAndVersionStatusForTask",
+				"build_id":   b.Id,
+				"task_id":    t.Id,
+				"status":     status,
+				"time_taken": t.TimeTaken,
+			})
 			return fmt.Errorf("error updating build: %v", err.Error())
```

Comment:
this really should be an errors.Wrap(err, "") situation

Reply:
Done.